### PR TITLE
withdraw token

### PR DESCRIPTION
**Pull request by PKUzjw**

### CODE REVIEW

File: src/components/home/elements/WithdrawToken.vue [hunk 1]
```diff
@@ -0,0 +1,763 @@
...
+                api: this.coldApi(),
+                opc: OPC_FUNCTION,
+                address: this.address,
+                // senderPublicKey: this.coldAddresses[this.address].publicKey,
+                fee: this.fee * VSYS_PRECISION,
+                feeScale: FEE_SCALE,
+                timestamp: Date.now(),
+                attachment: '',
+                contractId: this.contractId,
+                functionId: WITHDRAW_FUNCIDX,
```
Comment (SheldonYS) [inline src/components/home/elements/WithdrawToken.vue]: Only set `functionId` to constant is wrong. `functionId` will be changed for different type of token. (For split supported token, functionId is 7. But split unsupported token, it is 6)

File: src/constants.js [hunk 2]
```diff
@@ -46,7 +46,7 @@ export const DEFAULT_BASIC_CONFIG = {
     timeDiff: 0
 };
 
-export const NODE_IP = '/api'
+export const NODE_IP = 'api'
```
Comment (SheldonYS) [inline src/constants.js]: keep `NODE_IP` to "/api"

File: src/components/home/elements/TokenRecord.vue [hunk 3]
```diff
@@ -50,9 +50,11 @@
           <b-dropdown-item @click="burnToken">Destroy Token</b-dropdown-item>
           <b-dropdown-item v-if="enableStatus"
                            @click="splitToken">Split Token</b-dropdown-item>
-          <b-dropdown-item v-if="enableStatus && showUnsupportedFunction !== false"
+          <b-dropdown-item v-if="enableStatus && showUnsupportedFunction"
```
Comment (SheldonYS) [inline src/components/home/elements/TokenRecord.vue]: Remove this dropdown item. It is duplicated.

File: src/components/home/elements/WithdrawToken.vue [hunk 4]
```diff
@@ -0,0 +1,767 @@
...
+import { NODE_IP, CONTRACT_EXEC_FEE, TRANSFER_ATTACHMENT_BYTE_LIMIT, VSYS_PRECISION, FEE_SCALE, API_VERSION, PROTOCOL, OPC_ACCOUNT, OPC_FUNCTION, WITHDRAW_FUNCIDX, WITHDRAW_FUNCIDX_SPLIT } from '@/constants.js'
+import TokenConfirm from '../modals/TokenConfirm'
+import TokenSuccess from '../modals/TokenSuccess'
+import ColdSignature from '../modals/ColdSignature'
+import browser from '@/utils/browser'
+import common from '@/utils/common'
+import BigNumber from 'bignumber.js'
+import transaction from '@/utils/transaction'
+export default {
+    name: 'IssueToken',
```
Comment (Haynar216) [inline src/components/home/elements/WithdrawToken.vue]: Wrong name

File: src/components/home/elements/WithdrawToken.vue [hunk 5]
```diff
@@ -0,0 +1,767 @@
...
+            if (tabIndex === 0) {
+                this.resetPage()
+                this.pageId = 1
+            } else {
+                this.resetPage()
+                this.coldPageId = 1
+            }
+            this.scanShow = false
+        },
+        isAmountValid(type) {
```
Comment (Haynar216) [inline src/components/home/elements/WithdrawToken.vue]: It does not use 'type',it should be removed

File: src/components/home/elements/WithdrawToken.vue [hunk 6]
```diff
@@ -0,0 +1,767 @@
...
+            return BigNumber(num)
+        },
+        coldApi: function() {
+            return API_VERSION
+        },
+        isValidIssuer: function(addr) {
+            return addr === this.issuer
+        },
+        isSubmitDisabled(type) {
+            return !(BigNumber(this.amount).isGreaterThan(0))
```
Comment (Haynar216) [inline src/components/home/elements/WithdrawToken.vue]: these functions are not enough, please refer to functions in SplitToken

**Comment by SheldonYS** [comment]
some methods, for example, `onDecode`, `hideQrScan`, are useless. Please remove.

File: src/components/home/elements/WithdrawToken.vue [hunk 7]
```diff
@@ -0,0 +1,696 @@
...
+        isNumFormatValid(amount) {
+            return common.isNumFormatValid(amount)
+        },
+        isNegative(amount) {
+            return BigNumber(amount).isLessThan(0)
+        },
+        checkPrecision(amount) {
+            return common.checkPrecision(BigNumber(amount).multipliedBy(this.tokenUnity), 0)
+        },
+        isExceededMaxSupply(amount) {
```
Comment (SheldonYS) [inline src/components/home/elements/WithdrawToken.vue]: This is balance checking. Withdraw should not have balance checking.

File: src/components/home/elements/WithdrawToken.vue [hunk 8]
```diff
@@ -0,0 +1,631 @@
...
+                         @get-signature="getSignature"
+                         @next-page="coldNextPage"
+                         @prev-page="coldPrevPage"></ColdSignature>
+        </b-container>
+        <b-container v-show="coldPageId===4">
+          <TokenConfirm :address="address"
+                        :amount=inputAmount(amount)
+                        :fee="fee"
+                        :contract-id="contractId"
+                        :tx-type="'Wuthdraw Token from Contract'">
```
Comment (Haynar216) [inline src/components/home/elements/WithdrawToken.vue]: it should be 'Withdraw'

File: src/components/home/elements/WithdrawToken.vue [hunk 9]
```diff
@@ -0,0 +1,631 @@
...
+            } else {
+                this.coldPageId--
+            }
+        },
+        resetPage: function() {
+            this.amount = BigNumber(0)
+            this.pageId = 1
+            this.coldPageId = 1
+            this.sendError = false
+            this.coldSignature = ''
```
Comment (Haynar216) [inline src/components/home/elements/WithdrawToken.vue]: you should reset this.contractId variable

**Pull request merged**